### ISSUE TITLE
False positive: Unnecessary parentheses for functions with two function parameters

### ISSUE THREAD

**Issue by rubengees** [arturbosch/detekt#1051]
I have a function (inline and extension if that matters) with a signature like this:

```kt
fun test(first: (String) -> String, second: (String) -> String) {}
```

This method is called like that:

```kt
test({ something }) { something else }
```

Detekt reports:

```
detekt.UnnecessaryParentheses: Parentheses around the lambda ({ something})
                               are unnecessary and can be removed.
```

which is not the case if I have not overlooked something.
